app/src/main/java/com/example/fitjournal/journalEntry/components/JournalEntryList.kt

```diff
@@ -0,0 +1,86 @@
+package com.example.fitjournal.journalEntry.components
+
+import androidx.compose.foundation.ExperimentalFoundationApi
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.lazy.LazyColumn
+import androidx.compose.foundation.lazy.itemsIndexed
+import androidx.compose.material3.HorizontalDivider
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.material3.TextButton
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.res.stringResource
+import com.example.fitjournal.core.presentation.commoncomponents.listHeader.CategoryHeader
+import com.example.fitjournal.core.presentation.model.enums.WorkoutTypeEnum
+import com.example.fitjournal.core.presentation.theme.Spacing
+import com.example.fitjournal.journalEntry.domain.WorkoutDetail
+import com.example.fitjournal.library.presentation.screen.library.components.ExerciseItem
+
+@OptIn(ExperimentalFoundationApi::class)
+@Composable
+fun JournalEntryList(workoutList: List<WorkoutDetail>, selectedWorkout: (WorkoutDetail) -> Unit) {
+    val listOfWeightLiftingWorkouts = workoutList.filter { it.workoutType == WorkoutTypeEnum.WEIGHT_TRAINING }
+    val listOfCardioWorkouts = workoutList.filter { it.workoutType == WorkoutTypeEnum.CARDIO }
+    val listOfCalisthenicsWorkouts = workoutList.filter { it.workoutType == WorkoutTypeEnum.CALISTHENICS }
+
+    LazyColumn(
+        modifier = Modifier
+            .fillMaxWidth()
+            .padding(bottom = Spacing.spacing8)
+    ) {
+        if (listOfWeightLiftingWorkouts.isNotEmpty()) {
+            stickyHeader {
+                CategoryHeader(text = stringResource(id = WorkoutTypeEnum.WEIGHT_TRAINING.workoutTitle()))
+            }
+
```

Comment:
I think if we pass in a List<Pair<List<WorkoutDetail>, WorkoutType>> then we can just iterate through this list and for each non empty child list display header and list of items

Reply:
hmmm This is new to me.
how can I use it in Combine?

Reply:
something like this?
`
    val listOfWeightLiftingWorkouts = workoutList.filter { it.type == WorkoutTypeEnum.WEIGHT_TRAINING.workoutTitle() }
    val listOfCardioWorkouts = workoutList.filter { it.type == WorkoutTypeEnum.CARDIO.workoutTitle() }
    val listOfCalisthenicsWorkouts = workoutList.filter { it.type == WorkoutTypeEnum.CALISTHENICS.workoutTitle() }

    val something1  = Pair(WorkoutTypeEnum.WEIGHT_TRAINING, listOfWeightLiftingWorkouts)
    val something2  = Pair(WorkoutTypeEnum.CARDIO, listOfCardioWorkouts)
    val something3  = Pair(WorkoutTypeEnum.CALISTHENICS, listOfCalisthenicsWorkouts)

    val myNewList = listOf(something1, something2, something3)
`

but then how do you iterate in a Composable view?

Reply:
myNewList.foreach{ childList ->
if (childList.isNotEmpty()){
     sticky header (workoutType)
       WorkoutList
   }
}